fury/interactor.py

```diff
@@ -142,25 +147,84 @@ def _process_event(self, obj, evt):
             self.on_mouse_wheel_forward(obj, evt)
         elif evt == "MouseWheelBackwardEvent":
             self.on_mouse_wheel_backward(obj, evt)
+        elif evt == "DoubleClickEvent":
+            self.on_double_click(obj, evt)
 
         self.event.reset()  # Event fully processed.
 
+    # def on_left_button_down(self, _obj, evt):
+    #     self.left_button_down = True
+    #     prop = self.get_prop_at_event_position()
+    #     if prop is not None:
+    #         self.selected_props["left_button"].add(prop)
+    #         self.propagate_event(evt, prop)
+
+    #     if not self.event.abort_flag:
+    #         self.trackball_camera.OnLeftButtonDown()
+
+    def on_left_button_up(self, _obj, evt):
+        self.left_button_down = False
+        self.propagate_event(evt, *self.selected_props["left_button"])
+        self.selected_props["left_button"].clear()
+        self.trackball_camera.OnLeftButtonUp()
+
     def on_left_button_down(self, _obj, evt):
         self.left_button_down = True
+        self.nb_left_clicks += 1
+        prop = self.get_prop_at_event_position()
+
+        if self.nb_left_clicks == 1:
+            self.initial_state = self.trackball_camera.GetInteractor().GetEventPosition()
+            if self.initial_state == self.click_history:
+                # stop single click event here...
+                self.event.abort()
+                print("Double Clicked Detected. [Aborts previous single click]")
+                self.nb_left_clicks = 0
+                if prop is not None:
+                    prop.InvokeEvent(self.DoubleClickEvent, 'DoubleClickEvent')
```

Comment:
What you want here is: `self.propagate_event("DoubleClickEvent", prop)`.

Also, from my experience, the proper way to call `InvokeEvent` is `prop.InvokeEvent('DoubleClickEvent')`. But use our centralized method `self.propagate_event`.